contracts/Payroll.sol

```diff
@@ -335,7 +335,22 @@ contract Payroll is Initializable, OwnableUpgradeable, ReentrancyGuardUpgradeabl
      */
     function _performMultiPayment(Payment[] calldata _payments) internal {
         for (uint256 i = 0; i < _payments.length; i++) {
-            _performPayment(_payments[i].token, _payments[i].receivers, _payments[i].amountsToTransfer);
+            require(_payments[i].amountsToTransfer.length > 0, "Payroll: No amounts to transfer");
+            require(
+                _payments[i].amountsToTransfer.length == _payments[i].receivers.length,
+                "Payroll: Arrays must have same length"
+            );
+
+            if (_payments[i].token != address(0)) {
+                _performERC20Payment(_payments[i].token, _payments[i].receivers, _payments[i].amountsToTransfer);
+            } else {
+                _performETHPayment(_payments[i].token, _payments[i].receivers, _payments[i].amountsToTransfer);
+                uint256 leftOver = address(this).balance;
+                if (leftOver > 0) {
```

Comment:
You may have more than one payment in ETH, if this happens left over will throw an error the second time it's called.
Also we don't know if in the future we are going to hold ETH on this contract.
So we should get the msg.value then in the else substract the eth sent in  _performETHPayment
Finally after ciclying we should ask if leftOver is bigger than 0 then sent it.
Pseudo code would look like
```
leftOver = msg.value
for () {
   if (_payments[i].token != address(0)) {
     substracted = _performETHPayment()
     leftOver = leftOver - substracted
   }
}
if(leftOver > 0) {
  sendEth(leftOver)
}
```

Reply:
fixed